apps/aehttp/src/aehttp_dispatch_int.erl

```diff
@@ -649,3 +672,57 @@ get_block_range(GetFun, Req) when is_function(GetFun, 0) ->
 
     end.
 
+get_account_balance_at_hash(AccountPubkey, Hash) ->
+    {value, Trees} = aec_db:find_block_state(Hash),
+    AccountsMPTree = aec_trees:accounts(Trees),
+    case aec_accounts_trees:lookup(AccountPubkey, AccountsMPTree) of
+        none ->
+            {error, account_not_found};
+        {value, Account} ->
+            {ok, aec_accounts:balance(Account)}
+    end.
+
+-spec get_block_hash_optionaly_by_hash_or_height(map()) ->
+    {ok, binary()} | {error, not_found | invalid_hash | blocks_mismatch}.
+get_block_hash_optionaly_by_hash_or_height(Req) ->
+    GetHashByHeight =
+        fun(Height) ->
+            case aec_conductor:get_header_by_height(Height) of 
+                {error, chain_too_short} ->
+                    {error, not_found};
+                {ok, Header} ->
+                    {ok, _Hash} = aec_headers:hash_header(Header) 
+            end
+        end,
+    case {maps:get('height', Req), maps:get('hash', Req)} of
```

Comment:
gives an exception if not there, use maps:get/3

Reply:
A test should have caught this... why did it slip through?

Reply:
Actually this is a swagger thingy. All params that are not required are put in the Req (having a value `undefined`), even if not provided in the request. Using a maps:get/3 would never have the default param being used.

Reply:
Ok, that happens under water... good to know